scripts/launch_workflow_cf/README.md

```diff
@@ -1,28 +1,122 @@
+
+## Configure the Cloud and Terra execution environments
+
+1. Create or identify the Google Cloud Platform project where this Cloud Function will live / be billed.
+    1. Enable the [Cloud Functions API](https://console.developers.google.com/apis/library/cloudfunctions.googleapis.com) for that project.
+    1. Enable the [Secret Manager API](https://console.cloud.google.com/marketplace/product/google/secretmanager.googleapis.com) for that project.
+    1. Enable the [Cloud Build API](https://console.developers.google.com/apis/library/cloudbuild.googleapis.com) for this project.
+1. Create or identify the bucket in that project that will be used for files to trigger the Cloud Function.
+1. Create a new [service account](https://console.cloud.google.com/iam-admin/serviceaccounts) within that project. It will look like `<your service account name>@<your project id>.iam.gserviceaccount.com`.
+1. Download a JSON key for the newly created service account and store it in a safe place.
+1. Upload the JSON key to [Secret Manager](https://console.cloud.google.com/security/secret-manager).
+1. For the newly created secret, in column "Actions", choose "Copy Resource ID". You'll need this value for the deployment step and it will look like: `projects/<your project number>/secrets/<your secret name>/versions/1`.
+1. Give the service account the needed permissions to items within the project.
+    1. Grant role `Secret Manager Secret Accessor` to the service account.
+    1. Grant role `Storage Object Viewer` to the service account on the bucket that will trigger the Cloud Function.
+1. [Register the service account](https://github.com/broadinstitute/terra-tools/tree/master/scripts/register_service_account) in Terra.
+1. [Create a new Terra group](https://app.terra.bio/#groups) to hold only the service account. It will look like `
+<your group name>@firecloud.org`.
+1. Give the Terra Group the needed permissions to items within the project.
+    1. Add the Terra group to the billing project (a.k.a. workspace namespace).
+    1. Grant role `Storage Object Viewer` to the Terra group on the bucket that will trigger the Cloud Function. This is because, presumably, the workflow will need to read the contents of the file that triggered the workflow.
+    1. If your workflow uses a private Google Container Repository image, also grant role `Storage Object Viewer` to the Terra group on the bucket that holds the image (e.g., `artifacts.<your project id>.appspot.com`).
+1. Give the Terra Group the needed permissions within Terra.
+    1. Share the Terra workspace with the Terra group, granting `Writer` and `Can compute` access.
+    1. If the workspace has an Authorization Domain, ensure that the Terra group is a member of the Authorization Domain.
+1. If the workflow method is private, share the workflow method. Note that for the Broad Methods repository, sharing with groups is not yet supported. Instead share the method with the service account.
```

Comment:
in this case would you need to share it with the proxy as well? I'm really not sure, I'm still learning when the pet is used for permissions vs when the direct user is used.

Reply:
It appears that the answer is "no" because it works fine shared only with the service account user.